# TSSplorer - A tool for comparing and visualizing predicted transcription start sites in prokaryotes

## What is a transcription start site (TSS) and what do we need it for?
During prokaryotic transcription an RNA strand is synthesized by the protein complex RNA polymerase from 5' to 3' end.  
In the first main phase of transcription, the initiation, the RNA polymerase complex is assembled at the promotor. After assembly the RNA polymerase escapes the promotor and begins forming a strand of complementary bases to the template DNA. The first base to be incorporated is called the transcription start site (TSS). 

Therefore the TSS is the first transcribed base on the DNA template after the promoter as well as the first base in the transcribed RNA strand.  

Studying the TSS of prokaryotes is interesting for several reasons:

- Promoter recognition: The TSS is located near the promoter region. Identifying the TSS helps recongizing important elements of the promoter region.
- Gene annotation and accurate gene mapping: The TSS is essential for correctly annotating the start of genes. 
- Regulatory mechanisms: Many transcription factors bind near the TSS to either repress or enhance transcription. Identifying the TSS helps in mapping these binding sites and understanding their regulatory roles. The region between the TSS and the start codon, the 5' untranslated region, can play important roles in the regulation of translation. Knowing the TSS helps analyzing this region.

More on the biological backgrounds of transcription in prokaryotes in the [wiki](https://github.com/Integrative-Transcriptomics/tss-prediction-comparison/wiki/Prokaryotic-gene-structure-and-transcription).


## What distinguishes TSSplorer from other TSS prediction tools like TSSpredator?
## How to use TSSplorer?
## Workflow
## Languages and tools used in this project
- Visualization: Plotly
- Frontend: 
  - Language: JavaScript
  - Framework: React
- Backend: 
  - Python
## Contact
If you encounter any problems using TSSplorer please feel free to create an issue or contact us via email: 



